File: 设计模式/程序/2-AbstractFactoryPattern/AbFactoryPattern(python)/AbstractFactoryPatternDemo.py
from AbstractFactory import *

# 创建工厂生产者
factoryProducer = FactoryProducer()

# 创建形状工厂
shapeFactory = factoryProducer.GetFactory("Shape")
# 创建颜色工厂
colorFactory = factoryProducer.GetFactory("Color")
# 创建笔工厂
penFactory = factoryProducer.GetFactory("Pen")

# 使用工厂
circle = shapeFactory.GetShape("Circle")
circle.draw()
red = colorFactory.GetColor("Red")
red.fill()

File: 设计模式/程序/2-AbstractFactoryPattern/AbFactoryPattern(C#)/AbstractFactoryPatternDemo.cs
using System;


namespace AbstractFactoryPattern
{
    class Program
    {
        static void Main(string[] args)
        {
            FactoryMaker factoryMaker = new FactoryMaker();
            AbstractFactory colorFactory = factoryMaker.GetFactory("COLOR");
            Color red = colorFactory.GetColor("RED");
            red.fill();

            AbstractFactory shapeFactoty = factoryMaker.GetFactory("SHAPE");
            Shape rectangle = shapeFactoty.GetShape("RECTANGLE");
            rectangle.draw();
        }
    }
}